crates/neon-runtime/src/napi/object.rs

```diff
@@ -8,6 +8,7 @@ pub unsafe extern "C" fn new(out: &mut Local, env: Env) {
     napi::create_object(env, out as *mut _);
 }
 
+#[cfg(feature = "napi-6")]
```

Comment:
My only concern with all these changes is whether it’ll make it intimidating for maintenance, since it’s not obvious what N-API version any given API was introduced in. Is it documented directly in the standard N-API docs in a way that’s easy to discover? Maybe we could explain somewhere in a comment how to know which `napi-<N>` flag to use when adding an API?

Reply:
I left a [comment](https://github.com/neon-bindings/neon/pull/649/files#diff-bf24afa304665f35997079e36185410c84b2a375828d0399f592dce91767fa47R1-R4) in the `functions.rs` file to consult the N-API documentation for version numbers. This should make it very clear when adding a new N-API method.

However, that does not fully address this concern since we only test the latest N-API version. I filed a follow-up issue to fix this in CI: https://github.com/neon-bindings/neon/issues/659